src/libpriv/rpmostree-rpm-util.h

```diff
@@ -63,27 +63,9 @@ void
 rpmhdrs_diff_prnt_block (gboolean changelogs, struct RpmHeadersDiff *diff);
 
 /* Define cleanup functions for librpm here */
-static inline void
-rpmostree_cleanup_rpmheader (Header *h)
-{
-  if (*h)
-    headerFree (*h);
-}
-#define _cleanup_rpmheader_ __attribute__((cleanup(rpmostree_cleanup_rpmheader)))
-static inline void
-rpmostree_cleanup_rpmfi (rpmfi *fi)
-{
-  if (*fi)
-    rpmfiFree (*fi);
-}
-#define _cleanup_rpmfi_ __attribute__((cleanup(rpmostree_cleanup_rpmfi)))
-static inline void
-rpmostree_cleanup_rpmts (rpmts *tsp)
-{
-  if (*tsp)
-    rpmtsFree (*tsp);
-}
-#define _cleanup_rpmts_ __attribute__((cleanup(rpmostree_cleanup_rpmts)))
+G_DEFINE_AUTO_CLEANUP_FREE_FUNC(Header, headerFree, NULL)
+G_DEFINE_AUTO_CLEANUP_FREE_FUNC(rpmfi, rpmfiFree, NULL)
+G_DEFINE_AUTO_CLEANUP_FREE_FUNC(rpmts, rpmtsFree, NULL)
```

Comment:
One thing that's [bit people in the past](https://bugs.freedesktop.org/show_bug.cgi?id=95065#c5) is - this breaks when the library itself then later adds defines for cleanup macros.  Now...currently librpm doesn't depend on GLib, but it *could* do a trick like defining the macro if it detects `#include <glib.h>` was already done.

In practice...we should likely stick this in libdnf (and make more use of them there)  But I'm fine with doing it here for now.  Just add a `/* TODO: Move to libdnf */` maybe?